src/NexxLogic.BlobConfiguration.AspNetCore/FileProvider/EnhancedBlobChangeToken.cs

```diff
@@ -0,0 +1,233 @@
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+using System.Collections.Concurrent;
+using System.Security.Cryptography;
+
+namespace NexxLogic.BlobConfiguration.AspNetCore.FileProvider;
+
+internal class EnhancedBlobChangeToken : IChangeToken, IDisposable
+{
+    private readonly BlobServiceClient _blobServiceClient;
+    private readonly string _containerName;
+    private readonly string _blobPath;
+    private readonly TimeSpan _debounceDelay;
+    private readonly bool _useContentBasedDetection;
+    private readonly int _maxContentHashSizeMb;
+    private readonly ConcurrentDictionary<string, string> _contentHashes;
+    private readonly ConcurrentDictionary<string, Timer> _debounceTimers;
+    private readonly ILogger _logger;
+
+    private readonly CancellationTokenSource _cts;
+    private volatile bool _hasChanged;
+    private readonly object _lock = new object();
+    private readonly List<(Action<object?> callback, object? state)> _callbacks = new();
+
+    public bool HasChanged => _hasChanged;
+    public bool ActiveChangeCallbacks => true;
+
+    public EnhancedBlobChangeToken(
+        BlobServiceClient blobServiceClient,
+        string containerName,
+        string blobPath,
+        TimeSpan debounceDelay,
+        bool useContentBasedDetection,
```

Comment:
A cleaner API would be to use an enum: 

- content based
- etag based

That would clarify instantly what the detection strategy will be. With only the boolean flag, it's not clear to the user that the etag strategy would be used.

Even better would be to have an IChangeDetectionStrategy interface with two implementations: one for content based detection and one for etag detection. This would also make things clear, and extensible.

It might seem overkill for just two strategies, but you'll be happily surprised how quickly it cleans up the change token class :)